flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CommonTable.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import org.apache.flink.table.api.TableSchema;
+
+import java.util.Map;
+
+/**
+ * CommonTable is the common parent of table and view. It has a map of
+ * key-value pairs defining the properties of the table.
+ */
+public interface CommonTable {
+	/**
+	 * Get the properties of the table.
+	 * @return table property map
+	 */
+	Map<String, String> getProperties();
+
+	/**
+	 * Get the schema of the table.
+	 * @return schema of the table
+	 */
+	TableSchema getSchema();
+
+	/**
+	 * Get a deep copy of the CommonTable instance.
+	 * @return a copy of the CommonTable instance
+	 */
+	CommonTable copy();
```

Comment:
Same comment as for database. Please add methods for a description and detailed description.

Reply:
Will do when comment on database is clarified.